src/terms/xc.jl

```diff
@@ -163,44 +163,47 @@ end
     isnothing(term.ρcore) && return nothing
 
     Vxc_real = xc_potential_real(term, basis, ψ, occupation; ρ, τ).potential
-    # TODO: move arrays to the CPU to enable forces calculations on the GPU.
-    #       Might require optimizations in the future.
     if basis.model.spin_polarization in (:none, :spinless)
-        Vxc_fourier = to_cpu(fft(basis, Vxc_real[:,:,:,1]))
+        Vxc_fourier = fft(basis, Vxc_real[:,:,:,1])
     else
-        Vxc_fourier = to_cpu(fft(basis, mean(Vxc_real, dims=4)))
+        Vxc_fourier = fft(basis, mean(Vxc_real, dims=4))
     end
 
-    model = basis.model
-    form_factors = atomic_density_form_factors(basis, CoreDensity())
+    form_factors, iG2ifnorm = atomic_density_form_factors(basis, CoreDensity())
     nlcc_groups = [(igroup, group) for (igroup, group) in enumerate(basis.model.atom_groups)
-                   if has_core_density(model.atoms[first(group)])]
+                   if has_core_density(basis.model.atoms[first(group)])]
```

Comment:
Maybe use `filter` here ... and then only add the `igroup` in the `for` loop in `_forces_xc` below. I find it a little strange that the index is added, but this is not really used for anything in this function, but only in the called function. That's a little surprising.